src/BigIntFromString.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @since 0.5.11
+ */
+import * as t from 'io-ts'
+import { either } from 'fp-ts/lib/Either'
+import { NonEmptyString } from './NonEmptyString'
+
+/**
+ * @since 0.5.11
+ */
+export interface BigIntFromStringC extends t.Type<bigint, string, unknown> {}
+
+/**
+ * @example
+ * import { BigIntFromString } from 'io-ts-types/lib/BigIntFromString'
+ * import { right } from 'fp-ts/lib/Either'
+ * import { PathReporter } from 'io-ts/lib/PathReporter'
+ *
+ * assert.deepStrictEqual(BigIntFromString.decode('1'), right(BigInt(1)))
+ * assert.deepStrictEqual(PathReporter.report(BigIntFromString.decode('1.1')), ['Invalid value "1.1" supplied to : BigIntFromString'])
+ * assert.deepStrictEqual(PathReporter.report(BigIntFromString.decode('')), ['Invalid value "" supplied to : BigIntFromString'])
+ *
+ * @since 0.5.11
+ */
+export const BigIntFromString: BigIntFromStringC = new t.Type<bigint, string, unknown>(
+  'BigIntFromString',
+  // tslint:disable-next-line
```

Comment:
I'm not fond of this complete-disabling of tslint on this line, but I couldn't get [this rule](https://palantir.github.io/tslint/rules/typeof-compare/) to stop complaining. I took inspiration from [this io-ts commit](https://github.com/gcanti/io-ts/pull/417/files#diff-f41e9d04a45c83f3b6f6e630f10117feR374)